File: account-svc/src/main/java/club/hosppy/account/model/Account.java
package club.hosppy.account.model;

import lombok.*;
import org.hibernate.Hibernate;
import org.hibernate.annotations.GenericGenerator;
import org.springframework.util.StringUtils;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.Instant;
import java.util.Objects;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Account {

    @Id
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @GeneratedValue(generator = "uuid")
    private String id;

    private String name;

    private String email;

    private boolean confirmedAndActive;

    private Instant memberSince;

    private String phoneNumber;

    private String photoUrl;

    private boolean support;

    public boolean hasEmail() {
        return StringUtils.hasText(email);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Account account = (Account) o;
        return id != null && Objects.equals(id, account.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: src/main/kotlin/club/hosppy/account/model/Account.kt
package club.hosppy.account.model

import org.hibernate.Hibernate
import java.time.Instant
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
    var name: String = ""
    var email: String = ""
    var confirmedAndActive: Boolean = false
    var memberSince: Instant = Instant.now()
    var phoneNumber: String? = null
    var avatarUrl: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        val account = other as Account
        return id != null && id == account.id
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }

}